### PR TITLE
feat(tekton): expose Tekton Triggers trigger API

### LINKED ISSUE #2938: Impossible to define standalone Tekton Trigger
Currently the only way to create a TektonTrigger is to do it via EventListener object. This is not sufficient, because with the ability to use namespace selectors within EventListeners we would not need to the register it in an EventListener and the EL would be responsible for finding TektonTriggers it should handle.

Can we register a standalone Tekton Trigger today? I don't see anything exposed in the `tektonClient.v1alpha1()`. Am I missing something? Or it's something we lack suport for?

Possibly related to #2742.

### PR DESCRIPTION
**Pull request by goldmann**
## Description

Makes it possible to create and manage Tekton Triggers
directly and not via EventListener only.

Fixes #2938

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [x] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: extensions/tekton/generator-triggers/cmd/generate/generate.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ func main() {
 	// the CRD List types for which the model should be generated
 	// no other types need to be defined as they are auto discovered
 	crdLists := map[reflect.Type]schemagen.CrdScope{
+		reflect.TypeOf(triggers.Trigger{}):                   schemagen.Namespaced,
+		reflect.TypeOf(triggers.TriggerList{}):               schemagen.Namespaced,
```
Comment (Fabian-K) [inline extensions/tekton/generator-triggers/cmd/generate/generate.go:33]: nit: it should be sufficient to include the *List

**Review by Fabian-K — APPROVED** [review]
Looks great!

File: extensions/tekton/generator-triggers/cmd/generate/generate.go (hunk 1 above)
Comment (goldmann) [reply extensions/tekton/generator-triggers/cmd/generate/generate.go:33]: I'm happy to change it!

Comment (goldmann) [reply extensions/tekton/generator-triggers/cmd/generate/generate.go]: Aaaand done!

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**